db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBPersistentSubscriptionsClient.java

```diff
@@ -164,4 +176,243 @@ public CompletableFuture deleteToAll(String group, DeletePersistentSubscriptionO
 
         return new SubscribePersistentSubscriptionToAll(this.client, group, options, listener).execute();
     }
+
+    public CompletableFuture<List<PersistentSubscriptionInfo>> listAll(ListPersistentSubscriptionsOptions options) {
+        return getHttpURLConnection(options, "/subscriptions").thenApply(http -> {
+            try {
+                http.setRequestMethod("GET");
+
+                throwOnError(http.getResponseCode());
+
+                String content = readContent(http);
+                List<PersistentSubscriptionInfo> ps = new ArrayList<>();
+
+                for (JsonNode node: mapper.readTree(content)) {
+                    ps.add(parseSubscriptionInfo(node));
+                }
+                return ps;
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            } finally {
+                http.disconnect();
+            }
+        });
+    }
+
+    public CompletableFuture<List<PersistentSubscriptionInfo>> listAll() {
+        return listAll(ListPersistentSubscriptionsOptions.get());
+    }
+
+    public CompletableFuture<List<PersistentSubscriptionInfo>> listForStream(String stream, ListPersistentSubscriptionsOptions options) {
+        return getHttpURLConnection(options, "/subscriptions/" + stream).thenApply(http -> {

```

Comment:
**FIX:** I think we can do it right now through gRPC since when this PR was merged: https://github.com/EventStore/EventStore/pull/3352.